plugins/woocommerce/woocommerce.php

```diff
@@ -60,3 +61,24 @@ function wc_get_container() {
 
 // Global for backwards compatibility.
 $GLOBALS['woocommerce'] = WC();
+
+// Jetpack's Rest_Authentication needs to be initialized even before plugins_loaded.
+if ( class_exists( \Automattic\Jetpack\Connection\Rest_Authentication::class ) ) {
+	\Automattic\Jetpack\Connection\Rest_Authentication::init();
+}
+
+/**
+ * Initialize the Jetpack functionalities: connection, identity crisis, etc.
+ */
+function wc_jetpack_init() {
+	$config = new Automattic\Jetpack\Config();
+	$config->ensure(
+		'connection',
+		array(
+			'slug' => 'woocommerce',
+			'name' => __( 'WooCommerce', 'woocommerce' ),
+		)
+	);
+}
+
+add_action( 'plugins_loaded', 'wc_jetpack_init', 1 );
```

Comment:
I think these type of initialization routines belong in `/includes/class-woocommerce.php`. It might be better to move it there? That code also runs early (see line 34)

Reply:
Good idea 👍 Updated in https://github.com/woocommerce/woocommerce/pull/38674/commits/d0e5c3476594ffe92452d410ca2502902e40ea7b